data-service/src/main/java/com/amazonaws/blox/dataservice/model/InstanceGroup.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.model;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import java.util.Set;
+
+@Value
+@Builder
+public class InstanceGroup {
+
+  @NonNull private String clusterArn;
```

Comment:
I think we'll have to  make Environments a subresource of a Cluster, like Services are today (i.e. and Environment is tied to one and only one Cluster for its lifetime). The way [clusters are defined](http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_clusters.html), I can't think of a reason why a user would change the cluster an Environment is targeted at. Additionally, I think that cluster-changing deployment would be hard to implement, since we'd then have to consider state from all clusters where an environment has ever been deployed in order to undeploy tasks that may have lingered on those clusters.

Maybe put it on Environment instead?

Reply:
I don't understand, how does having the cluster in the instance group prevent it from being immutable?

Reply:
It would be inconsistent at the API level to be able to change one part of the instance group, and not the other.

That's not a super strong argument though. I think being able to get environment by cluster is a better argument. If we indexed on cluster in this table, there would be no way to get just one record per environment when querying by cluster. And if we put it in both tables, then we can't atomically update it.